### ISSUE TITLE
Migrate refs to use React.createRef();

### ISSUE THREAD

**Issue by cleverbeagle** [cleverbeagle/pup#147]
Just found out about this. Minor detail but good to stay up to date with latest technique: https://reactjs.org/docs/refs-and-the-dom.html#creating-refs.

**Comment by merlinpatt** [comment]
Isn't this the new recommended pattern?

**Comment by cleverbeagle** [comment]
Yes and no from what I'm reading. The function ref style hasn't been deprecated, it's just being considered another way to do the same thing. Unless it's technically required I don't see a point to making the change—otherwise it's just pushing paper. 

Function refs are still documented here: https://reactjs.org/docs/refs-and-the-dom.html#callback-refs
